### PR TITLE
LPD-2294 [Upgrade process] Drop views of the object definition dynamic tables from partitioned DB

### PR DESCRIPTION
**Pull request by p-albuquerque**
Related issue: https://issues.liferay.com/browse/LPD-2294;

**NOTE:**
For upgrade process that perform DDL operations, integration tests aren't needed:
https://liferay.atlassian.net/wiki/spaces/ENGCOREUPGRADE/pages/1654719711/Upgrade+Development+Training#Testing

### CODE REVIEW

**Comment by p-albuquerque** [comment]
Thread opened about this:
https://liferay.slack.com/archives/C1SJ64S9X/p1725642335918789

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/upgrade/v9_2_2/SchemaUpgradeProcess.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ */
+public class SchemaUpgradeProcess extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		long companyId = CompanyThreadLocal.getCompanyId();
+		long defaultCompanyId = PortalInstancePool.getDefaultCompanyId();
+
+		if ((defaultCompanyId == companyId) ||
+			!DBPartition.isPartitionEnabled()) {
```
Comment (marcelabc) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/upgrade/v9_2_2/SchemaUpgradeProcess.java:31]: Can we sort the conditions?
**!DBPartition.isPartitionEnabled()** || (**companyId** == defaultCompanyId) ?

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi 
I finished my review. Let me know if you have questions :smile:

We can send our PR to the database team. I will check if this is necessary for you.

**Comment by achaparro** [comment]
@marcelabc @p-albuquerque see:
https://liferay.slack.com/archives/CUQP9C673/p1725878476216579?thread_ts=1725630491.655429&cid=CUQP9C673

**Pull request closed without merging**